index.d.ts

```diff
@@ -37,6 +44,6 @@ await pWaitFor(() => pathExists('unicorn.png'));
 console.log('Yay! The file now exists.');
 ```
 */
-export default function pWaitFor(condition: () => PromiseLike<boolean> | boolean, options?: Options): Promise<void>;
+export default function pWaitFor<ResolveValueType>(condition: (resolveWith: (value: ResolveValueType) => ResolveValue<ResolveValueType>) => PromiseLike<boolean> | boolean | ResolveValue<ResolveValueType> | PromiseLike<ResolveValue<ResolveValueType>>, options?: Options): Promise<ResolveValueType>;
```

Comment:
It seems like this type definition causes the `ResolveValueType` to always resolve to unknown:

![image](https://user-images.githubusercontent.com/36966635/162096972-c0261d52-84f6-4746-bfb8-734591cff84a.png)

I think I got it to work by making the condition callback a generic type:

```typescript
export default function pWaitFor<ResolveValueType>(condition: (resolveWith: <T>(value: T) => ResolveValue<T>) => PromiseLike<boolean> | boolean | ResolveValue<ResolveValueType> | PromiseLike<ResolveValue<ResolveValueType>>, options?: Options): Promise<ResolveValueType>;
```

![Screen Shot 2022-04-06 at 8 33 25 PM](https://user-images.githubusercontent.com/36966635/162097020-5b63af30-1dfd-46b9-ae1a-1d152e0b6974.png)